src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -813,6 +813,15 @@ void ShadingPoint::compute_shading_basis() const
     const Vector3d s = normalize(cross(sn, t));
     m_shading_basis.build(sn, s, t);
 
+    if (m_primitive_type == PrimitiveCurve3)
+    {
+        const Vector3d X = normalize(get_dpdu(0));
+        const Vector3d sn_curve = normalize(get_original_shading_normal());
```

Comment:
Some if not most of this work is redundant with what overloads of `Basis::build()` do.

Reply:
As I mentioned, curves requires a different shading basis to the other shapes, hence not sure how to use the ```Basis.build()``` overloads

Reply:
I would like to suggest maybe a flag for hair curves so that a different shading basis is calculated? I don't know the most elegant way to do this at the moment.